### ISSUE TITLE
Jovimetrix causing long load times of the comfyui manager

### ISSUE THREAD

**Issue by Gerkinfeltser** [Amorano/Jovimetrix#5]
Hi, I'd like to start out saying thank you so much for the great nodes before I get into my issue!

As much as I love your extension it though, it seems to be causing the comfyui manager & full interface to load very slowly. Either when reloading or starting the web interface afresh after the server is running, the manager & accompanying improvements often take a half minute or so to load.

That being said, I do have a slow system, however no other extension seems to cause such a delay. Without Jovimetrix enabled, the full interface loads in under 7 seconds. Please let me know if there is a way to help debug what may be causing the sluggishness.

Again, thank you again for sharing your tools!

**Comment by Amorano** [comment]
Thanks for the report. Can I ask how many node packs/total nodes you have installed. Seems to be an issue when the startup needs to parse the nodes for color when there are a large amount of nodes to actually load atm.

If you could even just give me a list of the packs you have, I can try to mirror the setup and see if I can figure out how to work around, but it may simply be a brute force problem.

**Comment by Gerkinfeltser** [comment]
Hi, so here is the list of what I have installed atm. With Jov disabled the interface loads in about 5 seconds. With it enabled, 25 seconds. Wondering if it might be helpful to leave only the manager enabled, then turn on Jov. Then add in others one by one. Perhaps that's your plan of attack as well. :D Anyway, thanks again!

`
cg-use-everywhere
cg_custom_core
comfy-image-saver
ComfyMath
ComfyUI-Custom-Scripts
ComfyUI-Image-Selector
ComfyUI-Impact-Pack
ComfyUI-Inspire-Pack
ComfyUI-Manager
ComfyUI-Marigold
ComfyUI_ADV_CLIP_emb
ComfyUI_Comfyroll_CustomNodes
comfyui_controlnet_aux
ComfyUI_essentials
ComfyUI_UltimateSDUpscale
comfy_mtb
Cup-ClipBoard
Derfuu_ComfyUI_ModdedNodes
efficiency-nodes-comfyui
rgthree-comfy
save-image-extended-comfyui
select_folder_path_easy
tiled_ksampler
was-node-suite-comfyui
clipseg.py
`

**Comment by Amorano** [comment]
Thank you for the list! 

The only ones I couldn't find to install in the test were: cg_custom_core, Cup-ClipBoard and clipseg.py (obviously), but I don't think that affected anything, it was easy to see.

Using the existing code with all the libraries they load slow.

I made an update to load the color panel itself after comfy load, not during, and that seems to have resolved the issue and it doesn't even slow the panel on open (just a slight 0.5 delay).

Cheers!

**Comment by Gerkinfeltser** [comment]
Hey, thank so much! This seems much improved. Full GUI loading went down to less than 7 seconds on my [slow] system. 👍 

(& fwiw the Cup-Clipboard was some junk from a Blender addon that I ended up never using so removed but left a lot of crap lying around. cg_custom_core is a common library type extension for things by @chrisgoringe like cg-noise, image-picker, and use-everywhere.)

**Comment by tastez** [comment]
im having this same issue now when using the midi node. Model loading and processing is 100 times slower. 
With or without recursive rgthree option

**Comment by Amorano** [comment]
> im having this same issue now when using the midi node. Model loading and processing is 100 times slower. With or without recursive rgthree option

You should [log this as an issue on my actual repository](https://github.com/Amorano/Jovimetrix), with maybe a screenshot of your setup and the enviornment/console logs.

**Comment by Amorano** [comment]
> im having this same issue now when using the midi node. Model loading and processing is 100 times slower. With or without recursive rgthree option

Can you give me a little more information? Screenshot, console?

I am using the node(s) right now, and they are running fine. I am using windows.

**Comment by tastez** [comment]
probably unrelated but still -
C```
:\Apps_2023\ComfyUI_windows_portable\ComfyUI\custom_nodes\Jovimetrix\sup\anim.py:326: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit(cache=True)
2024-08-08 19:39:25.941 | WARNING  | Jovimetrix:__init__:884 - module failed C:\Apps_2023\ComfyUI_windows_portable\ComfyUI\custom_nodes\Jovimetrix\core\create_glsl.py
2024-08-08 19:39:25.941 | WARNING  | Jovimetrix:__init__:885 - No module named 'glfw'
2024-08-08 19:39:26.202 | WARNING  | Jovimetrix.core.utility:<module>:51 - no gifski support
```

now the problem is seemingly only with the midi read object on my end.
I have Rgthree-comfy recursive mode toggled on
i have comfy manager installed.

if i have a simple setup like this
![image](https://github.com/user-attachments/assets/63ab1499-4bad-4391-af6c-647d70fdcf60)

i get this in console
"[rgthree] Using rgthree's optimized recursive execution.
Prompt executed in 0.11 seconds"

and fans start spinning 

not seeing any actual activity
![image](https://github.com/user-attachments/assets/62643f32-640a-412b-bc74-09f3ff669582)
task manager also not showing anything of note on both gpu or cpu.

putting down a ksampler and a model loader and executing the prompt leads to reallllly slow model load and processing.
there doesnt seem to be any stress, so not sure why its doing this. Normally my fans also dont just spin for no reason
![image](https://github.com/user-attachments/assets/4b5378ec-cae3-4648-a8be-ca2da52cd3e6)

**Comment by Amorano** [comment]
> probably unrelated but still -
> 
>   out = torch.nn.functional.scaled_dot_product_attention(q, k, v, attn_mask=mask, dropout_p=0.0, is_causal=False)
> 100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:02<00:00,  7.18it/s]
> Requested to load AutoencoderKL
> Loading 1 new model
> Prompt executed in 6.00 seconds
> ```

I will try to replicate this, but I would try to uninstall/reinstall given it says it cant find/use glfw. Are you using this on an actual machine, local, not a headless remote machine?

**Comment by Amorano** [comment]
> probably unrelated but still -
> 
> ```
> :\Apps_2023\ComfyUI_windows_portable\ComfyUI\custom_nodes\Jovimetrix\sup\anim.py:326: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
>   @jit(cache=True)
> 2024-08-08 19:39:25.941 | WARNING  | Jovimetrix:__init__:884 - module failed C:\Apps_2023\ComfyUI_windows_portable\ComfyUI\custom_nodes\Jovimetrix\core\create_glsl.py
> 2024-08-08 19:39:25.941 | WARNING  | Jovimetrix:__init__:885 - No module named 'glfw'
> 2024-08-08 19:39:26.202 | WARNING  | Jovimetrix.core.utility:<module>:51 - no gifski support
> ```
> 
> now the problem is seemingly only with the midi read object on my end. I have Rgthree-comfy recursive mode toggled on i have comfy manager installed.
> 
> if i have a simple setup like this ![image](https://private-user-images.githubusercontent.com/78642112/356326794-63ab1499-4bad-4391-af6c-647d70fdcf60.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxNDE0NDIsIm5iZiI6MTcyMzE0MTE0MiwicGF0aCI6Ii83ODY0MjExMi8zNTYzMjY3OTQtNjNhYjE0OTktNGJhZC00MzkxLWFmNmMtNjQ3ZDcwZmRjZjYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDE4MTkwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZmMmIyYmNiZjg4ZDI1NDkyOGU0MzYwYWE3Mzc4YzUxY2RmMjY5NjZhMDNkMmI4YWEwYjM2ZWI0YjkwM2E0NGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ucxHsIPfvL-hAWd_PzQE5eko7Ed9dcC42iMYbGNFPpk)
> 
> i get this in console "[rgthree] Using rgthree's optimized recursive execution. Prompt executed in 0.11 seconds"
> 
> and fans start spinning
> 
> not seeing any actual activity ![image](https://private-user-images.githubusercontent.com/78642112/356326893-62643f32-640a-412b-bc74-09f3ff669582.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxNDE0NDIsIm5iZiI6MTcyMzE0MTE0MiwicGF0aCI6Ii83ODY0MjExMi8zNTYzMjY4OTMtNjI2NDNmMzItNjQwYS00MTJiLWJjNzQtMDlmM2ZmNjY5NTgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDE4MTkwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUyNDgwYTg0ZjA3NDBjMDc0M2Y3YTZiNjdiMTYwM2Q0YThkNzFmZGM0Yzg4Y2E2ZWRlYWUyODcwMDU2NjY5ODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pXIceg0BN4ggExRP7HnzJ_M6H-06TAKt3c3TZUgNLs0) task manager also not showing anything of note on both gpu or cpu.
> 
> putting down a ksampler and a model loader and executing the prompt leads to reallllly slow model load and processing. there doesnt seem to be any stress, so not sure why its doing this. Normally my fans also dont just spin for no reason ![image](https://private-user-images.githubusercontent.com/78642112/356328201-4b5378ec-cae3-4648-a8be-ca2da52cd3e6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxNDE0NDIsIm5iZiI6MTcyMzE0MTE0MiwicGF0aCI6Ii83ODY0MjExMi8zNTYzMjgyMDEtNGI1Mzc4ZWMtY2FlMy00NjQ4LWE4YmUtY2EyZGE1MmNkM2U2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDE4MTkwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4NjZmOTVlYmQ2OWZiMzJhOGE3ODE5Njc0YmEwZGFkMGZmYzNlYzFmN2NiNGEyNzA0ZWYyZTNmZWIwMGM0ZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HDaIoNvHkaitPpBpswTwMHadHXY4PEUvAt0r4l_ZicU)
> 
> ** prompt is still executing - i think it says 21.5 minutes as an estimate ![image](https://private-user-images.githubusercontent.com/78642112/356329764-ee4165cf-6070-4cae-9130-ce61b249c3ca.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxNDE0NDIsIm5iZiI6MTcyMzE0MTE0MiwicGF0aCI6Ii83ODY0MjExMi8zNTYzMjk3NjQtZWU0MTY1Y2YtNjA3MC00Y2FlLTkxMzAtY2U2MWIyNDljM2NhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDE4MTkwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxNzRhOTc1YjcyY2Q0YWMxYzc5NWM4NjE3NGI3NzBmOTM2YWJmMTA4Njk0MDcyY2Q5NjM2NmU4ODA0MjAyN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.easGqrGa1bt-6KPp8JxiH0hgBryjQnyVrZn-t7NOUPg)
> 
> if i remove the midi node and hit execute
> 
> ```
> [rgthree] Using rgthree's optimized recursive execution.
> [rgthree] First run patching recursive_output_delete_if_changed and recursive_will_execute.
> [rgthree] Note: If execution seems broken due to forward ComfyUI changes, you can disable the optimization from rgthree settings in ComfyUI.
> model weight dtype torch.float16, manual cast: None
> model_type EPS
> Using xformers attention in VAE
> Using xformers attention in VAE
> Model doesn't have a device attribute.
> Model doesn't have a device attribute.
> loaded straight to GPU
> Requested to load SDXL
> Loading 1 new model
> Requested to load SDXLClipModel
> Loading 1 new model
> C:\Apps_2023\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\attention.py:407: UserWarning: 1Torch was not compiled with flash attention. (Triggered internally at ..\aten\src\ATen\native\transformers\cuda\sdp_utils.cpp:263.)
>   out = torch.nn.functional.scaled_dot_product_attention(q, k, v, attn_mask=mask, dropout_p=0.0, is_causal=False)
> 100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:02<00:00,  7.18it/s]
> Requested to load AutoencoderKL
> Loading 1 new model
> Prompt executed in 6.00 seconds
> ```

![image](https://github.com/user-attachments/assets/0f3a1f59-39b1-42c3-a635-7d10b3b876c4)

So its clear what I am getting. I am also using an MPK Mini =P

**Comment by tastez** [comment]
yep im getting that too! just adding actual ksamplers and etc makes stuff go into a time warp. I'll reinstall!

**Comment by Amorano** [comment]
> yep im getting that too! just adding actual ksamplers and etc makes stuff go into a time warp. I'll reinstall!

ok! I have a tweak or two I see for the midi loop itself, I will push that out here shortly as well.

Cheers for the ping!

**Comment by tastez** [comment]
thanks! It looks like reinstall solved the shader issue (sweeet) but the ksampler stuff is still a problem here. Will check it out after updating!

**Comment by tastez** [comment]
its fixed!!!!!!! this is so amazing thank you!
